ui/pages/settings/experimental-tab/experimental-tab.component.js

```diff
@@ -71,16 +71,50 @@ export default class ExperimentalTab extends PureComponent {
     handleSettingsRefs(t, t('experimental'), this.settingsRefs);
   }
 
+  ///: BEGIN:ONLY_INCLUDE_IN(blockaid)
+  toggleSecurityAlert(value) {
+    const { setSecurityAlertsEnabled, transactionSecurityCheckEnabled } =
+      this.props;
+    this.context.trackEvent({
+      category: MetaMetricsEventCategory.Settings,
+      event: 'Enabled/Disable security_alerts_enabled',
+      properties: {
+        action: 'Enabled/Disable security_alerts_enabled',
+        legacy_event: true,
+      },
+    });
+    setSecurityAlertsEnabled(!value);
+    if (!value && transactionSecurityCheckEnabled) {
+      this.toggleTransactionSecurityCheck(true);
+    }
+  }
+  ///: END:ONLY_INCLUDE_IN
+
+  toggleTransactionSecurityCheck(value) {
+    const { securityAlertsEnabled, setTransactionSecurityCheckEnabled } =
+      this.props;
+    this.context.trackEvent({
+      category: MetaMetricsEventCategory.Settings,
+      event: 'Enabled/Disable TransactionSecurityCheck',
+      properties: {
+        action: 'Enabled/Disable TransactionSecurityCheck',
+        legacy_event: true,
+      },
+    });
+    setTransactionSecurityCheckEnabled(!value);
```

Comment:
[suggestion] to help with readability and understanding of the code, I think it might be helpful to create a new const above

```
const newValue = !value;
```

comment applies to both toggle functions

Reply:
PR updated.